kcidb/monitor/spool/__init__.py

```diff
@@ -157,15 +170,21 @@ def pick(self, id, timestamp=None, timeout=None):
         def pick_if_not_picked(transaction, doc, timestamp):
             """Pick notification, if not picked yet"""
             # Get the document snapshot
-            snapshot = doc.get(field_paths=["picked_until", "message"],
+            snapshot = doc.get(field_paths=["picked_until", "message",
+                                            "time_to_send"],
                                transaction=transaction)
             picked_until = snapshot.get("picked_until")
             message_text = snapshot.get("message")
+            time_to_send = snapshot.get('time_to_send')
             # If the document doesn't exist, has no "picked_until" field,
             # no "message" field, or the picking has not timed out yet
             if not picked_until or not message_text or \
                picked_until > timestamp:
                 return None
+            # Check that notification is ready to be sent out
+            if time_to_send > datetime.datetime.now(datetime.timezone.utc):
```

Comment:
We need to be comparing to the `timestamp` argument here instead of the current time (see the argument docs). This would let us imitate picking at any time in testing, and also allow using consistent "current time" between multiple invocations of the function and other parts of the system.

Reply:
Instead of adding a separate if statement, do you think you could put the condition into the previous one? It does the same thing: refusing to pick the notification. It would get a lot of conditions, but they will all be doing the same thing.